src/components/timeline/timeline.component.less

```diff
@@ -0,0 +1,313 @@
+ux-timeline {
```

Comment:
We need to ensure that any classes designed to only target this component are wrapped in the component selector.

So in this case everything should be within the ux-timeline rule:

```less
ux-timeline {
   ... all styles here
}
```

This prevents any styles from affecting other elements with the same name.

There is also a lot of duplication here, as you will notice there are alot of `.timeline` rules. In Less we can nest all these rules so this:

```less
.timeline {
    &:after:extend(.hpe-chevron-down:before) {
        font-family: 'hpe-icons';
        position: absolute;
        color: @timeline-color;
        left: 56px;
        bottom: -9px;
        transform: translateX(-0.5px);
    }
}
  
.timeline {
    li {
        .timeline-badge {
            &.critical {
                background-color: @timeline-badge-critical !important;
            }
        }
    }
}
```

would become:

```less
.timeline {
    &:after:extend(.hpe-chevron-down:before) {
        font-family: 'hpe-icons';
        position: absolute;
        color: @timeline-color;
        left: 56px;
        bottom: -9px;
        transform: translateX(-0.5px);
    }

    li {
        .timeline-badge {
            &.critical {
                background-color: @timeline-badge-critical !important;
            }
        }
    }
}
```